### PR TITLE
Add parameter to check for httpOnly urls; Fixes #11

### LINKED ISSUE #11: Only allow HTTP URLs by default
```js
isAbsoluteUrl('javascript:throw%20document.cookie') // true
```

Although this might be technically correct (?), it seems like it's not what most people would think of as an absolute URL, and it could lead to a false sense of security that the input has been validated as a 'real' URL.

Possible solutions:

- Release a new major version that ensures the protocol is `http` or `https`
  - Maybe with an option like `{ allowNonHttp: true }` for people who want the old behaviour.
- Or just add a warning in the readme that this will return true for any protocol, even `javascript:`.

### PR DESCRIPTION
**Pull request by acagastya**
Add potentially breaking change to check for `http[s]` URLs by default as requested.

### CODE REVIEW

**Comment by sindresorhus** [comment]
This will also need docs and TypeScript type (following  https://github.com/sindresorhus/typescript-definition-style-guide) updates.

And give the PR a descriptive title.

**Comment by acagastya** [comment]
I will update the docs and types shortly.  I have updated the PR title.  Thanks for pointing it out.

**Comment by acagastya** [comment]
@sindresorhus I had updated `index.d.ts` by hovering over `module.exports` -- was it supposed to be done using any script?

File: test.js [hunk 1]
```diff
@@ -3,11 +3,17 @@ import isAbsoluteUrl from '.';
 
 test('main', t => {
 	t.true(isAbsoluteUrl('http://sindresorhus.com'));
+	t.true(isAbsoluteUrl('http://sindresorhus.com', {httpOnly: true}));
 	t.true(isAbsoluteUrl('https://sindresorhus.com'));
-	t.true(isAbsoluteUrl('httpS://sindresorhus.com'));
-	t.true(isAbsoluteUrl('file://sindresorhus.com'));
-	t.true(isAbsoluteUrl('mailto:someone@example.com'));
-	t.true(isAbsoluteUrl('data:text/plain;base64,SGVsbG8sIFdvcmxkIQ%3D%3D'));
+	t.true(isAbsoluteUrl('https://sindresorhus.com', {httpOnly: true}));
```
Comment (sindresorhus) [inline test.js:8]: The `false` option assertions should be in a separate test.

File: readme.md [hunk 2]
```diff
@@ -18,13 +20,23 @@ const isAbsoluteUrl = require('is-absolute-url');
...
+//=> true
+
 isAbsoluteUrl('//sindresorhus.com');
 //=> false
 
 isAbsoluteUrl('foo/bar');
 //=> false
 ```
 
+By default only `http` and `https` links are validated.  If you wish to check absolute URLs of protocols other than `http` (for example, `ftp:`, `mailto:`), use `isAbsoluteUrl(mayBeURL, { httpOnly: false })`.
```
Comment (sindresorhus) [inline readme.md:39]: These are not *links*.

File: readme.md [hunk 3]
```diff
@@ -18,13 +20,23 @@ const isAbsoluteUrl = require('is-absolute-url');
 isAbsoluteUrl('https://sindresorhus.com/foo/bar');
 //=> true
 
+isAbsoluteUrl('http://sindresorhus.com/foo/bar');
+//=> true
+
+isAbsoluteUrl('ftp://sindresorhus.com/foo/bar');
+//=> false
+
+isAbsoluteUrl('ftp://sindresorhus.com/foo/bar', { httpOnly: false });
```
Comment (sindresorhus) [inline readme.md:29]:
```suggestion
isAbsoluteUrl('ftp://sindresorhus.com/foo/bar', {httpOnly: false});
```

File: readme.md (hunk 2 above)
Comment (sindresorhus) [reply readme.md:39]: The option should have proper API docs like done here: https://github.com/sindresorhus/boxen#api

File: readme.md [hunk 4]
```diff
@@ -2,6 +2,8 @@
 
 > Check if a URL is absolute
 
+By default, it checks if the URL is of the `http` or `https` protocol.
```
Comment (sindresorhus) [inline readme.md:5]: I don't think "of" is correct here.

File: index.js [hunk 5]
```diff
@@ -12,5 +12,5 @@ module.exports = url => {
 
 	// Scheme: https://tools.ietf.org/html/rfc3986#section-3.1
 	// Absolute URL: https://tools.ietf.org/html/rfc3986#section-4.3
-	return /^[a-zA-Z][a-zA-Z\d+\-.]*:/.test(url);
+	return httpOnly ? /^https?:/.test(url) : /^[a-zA-Z][a-zA-Z\d+\-.]*:/.test(url);
```
Comment (sindresorhus) [inline index.js:15]:
```suggestion
	return httpOnly ? /^https?:\/\//.test(url) : /^[a-zA-Z][a-zA-Z\d+\-.]*:/.test(url);
```

File: index.d.ts [hunk 6]
```diff
@@ -17,6 +17,9 @@ isAbsoluteUrl('foo/bar');
 //=> false
 ```
 */
-declare function isAbsoluteUrl(url: string): boolean;
+declare function isAbsoluteUrl(
+	url: string,
+	{ httpOnly }?: { httpOnly?: boolean }
```
Comment (sindresorhus) [inline index.d.ts:22]: The options should be a separate type and have the same docs as the readme. Should follow: https://github.com/sindresorhus/typescript-definition-style-guide#documentation

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 6 above)
Comment (sindresorhus) [reply index.d.ts:22]: This is not fixed.

File: readme.md [hunk 7]
```diff
@@ -18,13 +20,41 @@ const isAbsoluteUrl = require('is-absolute-url');
...
+isAbsoluteUrl('ftp://sindresorhus.com/foo/bar', {httpOnly: false});
+//=> true
+
 isAbsoluteUrl('//sindresorhus.com');
 //=> false
 
 isAbsoluteUrl('foo/bar');
 //=> false
 ```
+By default only `http` and `https` links are validated.  If you wish to check absolute URLs of protocols other than `http` (for example, `ftp:`, `mailto:`), use `isAbsoluteUrl(mayBeURL, { httpOnly: false })`.
```
Comment (sindresorhus) [inline readme.md:39]: This should be in the `httpOnly` option docs. Not here.

File: test.js (hunk 1 above)
Comment (sindresorhus) [reply test.js:8]: Please don't mark as resolved when things are in fact no resolved.

**Comment by sindresorhus** [comment]
Bump

**Comment by sindresorhus** [comment]
Bump

**Pull request closed without merging**